### PR TITLE
Make service_account_email public, read-only

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: google/auth/app_engine.py [hunk 1]
```diff
@@ -87,6 +87,14 @@ def refresh(self, request):
         self.token, self.expiry = token, expiry
 
     @property
+    def service_account_email(self):
+        """The service account email."""
+        if self._service_account_id is not None:
+            return self._service_account_id
+        else:
+            return app_identity.get_service_account_name()
```
Comment (dhermes) [inline google/auth/app_engine.py]: Do you want to cache this on the instance?

File: google/auth/compute_engine/credentials.py [hunk 2]
```diff
@@ -93,6 +93,15 @@ def refresh(self, request):
             raise exceptions.RefreshError(exc)
 
     @property
+    def service_account_email(self):
+        """The service account email.
+
+        .. Note: This is not guarenteed to be set until :meth`refresh` has been
```
Comment (dhermes) [inline google/auth/compute_engine/credentials.py]: You are "guarenteed" to need a spellcheck here. Also do you need lowercase `note` for Sphinx or are the directives case-insensitive?

File: tests/compute_engine/test_credentials.py [hunk 3]
```diff
@@ -58,7 +60,7 @@ def test_refresh_success(self, get_mock, now_mock):
             datetime.datetime.min + datetime.timedelta(seconds=500))
 
         # Check the credential info
-        assert (self.credentials._service_account_email ==
+        assert (self.credentials.service_account_email ==
```
Comment (dhermes) [inline tests/compute_engine/test_credentials.py]: Can you do a `git grep '\._service_account_email'` for me?

Comment (jonparrott) [inline tests/compute_engine/test_credentials.py]:
```python
google/auth/compute_engine/credentials.py:        self._service_account_email = service_account_email
google/auth/compute_engine/credentials.py:            service_account=self._service_account_email)
google/auth/compute_engine/credentials.py:        self._service_account_email = info['email']
google/auth/compute_engine/credentials.py:                service_account=self._service_account_email)
google/auth/compute_engine/credentials.py:        return self._service_account_email
google/oauth2/service_account.py:        self._service_account_email = service_account_email
google/oauth2/service_account.py:            issuer=self._service_account_email,
google/oauth2/service_account.py:            subject=self._service_account_email)
google/oauth2/service_account.py:        return self._service_account_email
google/oauth2/service_account.py:            service_account_email=self._service_account_email,
google/oauth2/service_account.py:            service_account_email=self._service_account_email,
google/oauth2/service_account.py:            'iss': self._service_account_email,
system_tests/test_compute_engine.py:    assert credentials._service_account_email is not None
system_tests/test_service_account.py:    assert info['email'] == credentials._service_account_email
tests/test__oauth2client.py:    assert (new_credentials._service_account_email ==
tests/test__oauth2client.py:    assert (new_credentials._service_account_email ==
tests/test__oauth2client.py:    assert (new_credentials._service_account_email ==
```

Comment (dhermes) [inline tests/compute_engine/test_credentials.py]: Worth it to "fix" some of those?

File: google/auth/compute_engine/credentials.py (hunk 2 above)
Comment (jonparrott) [inline google/auth/compute_engine/credentials.py]: Spelling fixed, admonition made lower-case.

File: tests/compute_engine/test_credentials.py (hunk 3 above)
Comment (jonparrott) [inline tests/compute_engine/test_credentials.py]: Sure, most will stay though.

**Pull request merged**